### PR TITLE
Changes for GMD submission

### PR DESCRIPTION
**Pull request by samoliverowens**
- Fix bug when reading walltypes.inp when number of wall layers is greater than 3 so that Eastside demo can be run with 10 layers.
- Add switch for reading in scalar driver files so that DAPPLE validation case can be run without them.
- Include script for running the cases on ARCHER2.

### CODE REVIEW

File: docs/udales-namoptions-overview.md [hunk 1]
```diff
@@ -215,13 +215,14 @@ BCs at the bottom (BCbot; only effective if not covered with road facets): 1 = f
...
+| lscasrcr | .false. | .true., .false. |  Switch for 2-D network of point sources at lowest level as defined in scals.inp.xxx.  | - |
+| lscasrcl | .false. | .true., .false. |  Switch for passive scalar line source when using canyon geometry.  | - |
+| lscasrc | .false. | .true., .false. |  Switch for passive scalar point source defined by xS,yS,zS,SS,sigS. | - |
+| xS | 0 | `REAL` | Position of scalar source in x. | m |
+| yS | 0 | `REAL` | Position of scalar source in y. | m |
+| zS | 0 | `REAL` | Position of scalar source in z. | m |
+| SS | 0. | `REAL` | Strength of scalar source. | g/ms |
+| sigS | 0. | `REAL` | Standard deviation of scalar source. | m |
 
 ## Namelist DRIVER
```
Comment (bss116) [inline docs/udales-namoptions-overview.md:228]: can you add `lsdriver` please

Comment (bss116) [reply docs/udales-namoptions-overview.md:228]: and `nwalllayers` to ENERGYBALANCE

**Review by bss116 — COMMENTED** [review]
While we are at it, we should also update the sections for scalars and number of wall layers in our example simulations

File: docs/udales-namoptions-overview.md (hunk 1 above)
Comment (samoliverowens) [reply docs/udales-namoptions-overview.md:229]: Done - nwalllayers is already there

**Comment by samoliverowens** [comment]
> Isn't `lsdriver` under DRIVER? Yes sorry missed that nwalllayers was there already

Sorry yes, have moved it.

**Review by bss116 — APPROVED** [review]
Ok. changes are fine with me, please go ahead and squash-merge when you are ready.

**Pull request merged**